src/textfield/textfield.js

```diff
@@ -142,10 +142,12 @@
    * @public
    */
   MaterialTextfield.prototype.checkValidity = function() {
-    if (this.input_.validity.valid) {
-      this.element_.classList.remove(this.CssClasses_.IS_INVALID);
-    } else {
-      this.element_.classList.add(this.CssClasses_.IS_INVALID);
+    if (this.input_.validity) {
+      if (this.input_.validity.valid) {
```

Comment:
I did not think that the invalid class should added in else if validity was undefined - like this: this.input_.validity && this.input_.validity.valid
Should the logic be reversed like so: this.input_.validity && !this.input_.validity.valid -> add(invalid) so else would remove(invalid) rather that add it?